### PR TITLE
Родионов - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by RodionovF**

### CODE REVIEW

**Comment by RodionovF** [comment]
@KSTconsult @VlaKuz

File: code/segment_tree/model/segment_tree.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+            left_child = self.tree[vertex * 2]
+            right_child = self.tree[vertex * 2 + 1]
+
+            if self.type == 'sum':
+                self.tree[vertex] = left_child + right_child
+            elif self.type == 'min':
+                self.tree[vertex] = min(left_child, right_child)
+            elif self.type == 'max':
+                self.tree[vertex] = max(left_child, right_child)
```
Comment (VlaKuz) [inline code/segment_tree/model/segment_tree.py:27]: maybe it should be better to use dict with keys 'sum' 'min' 'max' and values with your operations?

File: code/segment_tree/model/test_segment_tree.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+class TestSegmentTree(unittest.TestCase):
+    def test_cannot_create_tree_with_specific_type(self):
+        with self.assertRaises(TypeError):
+            SegmentTree('exp')
+
+    def test_can_create_tree_only_with_list(self):
+        with self.assertRaises(TypeError):
+            array = {1: 2, 3: 4}
+
+            st = SegmentTree('sum')
```
Comment (VlaKuz) [inline code/segment_tree/model/test_segment_tree.py:15]: confusing name

**Comment by VlaKuz** [comment]
👍

**Comment by KSTconsult** [comment]
:+1:

**Comment by RodionovF** [comment]
@kirill-korniakov

**Pull request merged**